rhizome/postgres/lib/hugepages_setup.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative "../../common/lib/util"
+
+class HugepagesSetup
+  def initialize(instance)
+    @version, @cluster = instance.split("-", 2)
+  end
+
+  def hugepage_size_kib
+    r('awk \'/Hugepagesize/ {printf "%.2f\n", $2}\' /proc/meminfo').to_f
+  end
+
+  def get_postgres_param(param)
+    r "sudo -u postgres /usr/lib/postgresql/#{@version}/bin/postgres -D /dat/#{@version}/data -c config_file=/etc/postgresql/#{@version}/#{@cluster}/postgresql.conf -C #{param.shellescape}"
+  end
+
+  def shared_buffers_kib
+    shared_buffers_blocks = get_postgres_param("shared_buffers").to_f
+    block_size_bytes = get_postgres_param("block_size").to_f
+
+    shared_buffers_blocks * block_size_bytes / 1024
+  end
+
+  def stop_postgres_cluster
+    r "sudo pg_ctlcluster stop #{@version} #{@cluster}", expect: [0, 2] # 2 is "not running"
+  end
+
+  def hugepages_count
+    @hugepages_count ||= begin
+      shared_memory_size_kib = 1024 * get_postgres_param("shared_memory_size").to_f
+      (shared_memory_size_kib / hugepage_size_kib).ceil
```

Comment:
This is a trick for integral ceiling division that I have to look up every time. Granted, AI does make it faster. But I always know to reach for it :)

```suggestion
      (shared_memory_size_kib + hugepage_size_kib - 1) / hugepage_size_kib
```